### PR TITLE
Add test result case for SHOULD/MAY aspects of specs. Resolves #86

### LINKED ISSUE #86: Add a new test result state for SHOULD/MAY aspects of specifications
This should avoid Fails or N/As against things which an implementation may choose to implement. It should include a note to indicate why it may/may not be important to implement a given optional feature.

### PR DESCRIPTION
**Pull request by andrewbonney**
I'd prefer a different word to 'optional'. Suggestions welcome. Alternative two new TestResult states could be added for each of SHOULD and MAY, but that may be over the top.

Perhaps 'Not Implemented'?

### CODE REVIEW

File: TestResult.py [hunk 1]
```diff
@@ -38,3 +38,6 @@ def NA(self, detail):
 
     def FAIL(self, detail):
         return [self.description, "Fail", detail, self.name, self._time_elapsed()]
+
+    def OPTIONAL(self, detail):
+        return [self.description, "Optional", detail, self.name, self._time_elapsed()]
```
Comment (andrewbonney) [inline TestResult.py]:
```suggestion
        return [self.description, "Not Implemented", detail, self.name, self._time_elapsed()]
```

**Comment by garethsb-sony** [comment]
Yes, I'm not sure whether we want to distinguish

- between SHOULD not implemented vs. MAY not implemented (your point)
- between SHOULD/MAY fail vs. MUST fail (I don't think we do, a fail is a fail, even in an optional feature)
- between SHOULD/MAY pass vs. MUST pass (this one is interesting because highlighting those a little would act as a kind of *carrot* as opposed to the stick of "not implemented"?)

**Comment by andrewbonney** [comment]
That took me a few mental cycles - I think I'm still tired from last week. So you're suggesting a sort of 'extra green' pass state for the optional bits?

**Comment by peterbrightwell** [comment]
Maybe pass for an optional test could be a super-green that needs in BT.2020 or AdobeRGB while a compulsory test a regular green BT.709 or sRGB.

Well maybe not, but the carrot seems a good idea

**Comment by andrewbonney** [comment]
I had a quick go at this, but I think it needs a bit more thought, not least to indicate that it's possible to achieve this 'bonus' test result prior to running the test suite. I'll add another issue, but I'd suggest it's handled outside of this specific PR.

**Comment by andrewbonney** [comment]
I suspect the colour used for 'optional' elements will need to change so we can use this one for 'warning' (see https://github.com/AMWA-TV/nmos-testing/issues/101). The set we're choosing from suggests this one may have to end up at blue (bg-primary): https://getbootstrap.com/docs/4.0/utilities/colors/#background-color

**Comment by garethsb-sony** [comment]
I think the example Jasper gave for 'warning' seems similar enough to the SHOULD/MAY features we've been discussing. The statement about "pri" doesn't use normative language. I wouldn't have a problem using the same colour for both of these things, even if the phrase used were different?

**Comment by andrewbonney** [comment]
Ok. We can address that one in a similar way after this is merged in.

**Pull request merged**